libraries/BLE/src/BLEDevice.h

```diff
@@ -0,0 +1,414 @@
+     */
+    bool setTxPower(int txPower);
+    
+    /**
+     * @brief   Set advertising type as connectable/non-connectable
+     *
+     * @param[in]   connectable     true  - The device connectable
+     *                              false - The device non-connectable
+     *
+     * @return  none
+     *
+     * @note  Only for peripheral mode.
+     *         Default value is connectable
+     */
+    void setConnectable(bool connectable);
+
+    /**
+     * @brief   Set the value of the device name characteristic 
+     *
+     * @param[in] device  User-defined name string for this device.  Truncated if
+     *                   more than maximum allowed string length (20 bytes).
+     *
+     * @note This method must be called before the begin method
+     *       If device name is not set, a default name will be used
+     */
+    void setDeviceName(const char* deviceName);
+    
+    /**
+     * @brief   Set the appearance type for the BLE Peripheral Device
+     *
+     * See https://developer.bluetooth.org/gatt/characteristics/Pages/CharacteristicViewer.aspx?u=org.bluetooth.characteristic.gap.appearance.xml
+     * for available options.
+     *
+     * @param[in] appearance Appearance category identifier as defined by BLE Standard
+     *
+     * @return BleStatus indicating success or error
+     *
+     * @note This method must be called before the begin method
+     */
+    void setAppearance(unsigned short appearance);
+
+    /**
+     * @brief   Add a Service to the BLE Peripheral Device
+     *
+     * @param[in] attribute     The service that will add to Peripheral
+     *
+     * @return BLE_STATUS_T     Indicating success or error type
+     *
+     * @note This method must be called before the begin method
+     */
+    BLE_STATUS_T addService(BLEService& attribute);
+    
+    /**
+     * @brief   Construct the ADV data and start send advertisement
+     *
+     * @param   none
+     *
+     * @return  BLE_STATUS_T       0 - Success. Others - error code
+     *
+     * @note  none
+     */
+    BLE_STATUS_T startAdvertising();
+    
+    /**
+     * @brief   Stop send advertisement
+     *
+     * @param   none
+     *
+     * @return  none
+     *
+     * @note  none
+     */
+    void stopAdvertising();
+
+    /**
+     * @brief   Get currently connected central 
+     *
+     * @return BLEDevice Connected central device
+     *
+     * @note  Peripheral mode only
+     */
+    BLEDevice central();
+    
+    /**
+     * @brief   Get currently connected peripheral 
+     *
+     * @param   none
+     *
+     * @return  none
+     *
+     * @note  Central mode only. How to distinguish the peripheral?
+     */
+    BLEDevice peripheral();
+    
+    /**
+     * @brief   Release the resources when link lost
+     *
+     * @param   none
+     *
+     * @return  none
+     *
+     * @note  Peer devices only. Do nothing if local BLE device called.
+     */
+    void linkLost();
```

Comment:
This is an API I added. Now I let the resource release at deconstructor. I will delete it.